lib/chef/mixin/shell_out.rb

```diff
@@ -70,29 +71,29 @@ def shell_out_compact(*args, **options)
       end
 
       def shell_out_compact!(*args, **options)
+        options = maybe_add_timeout(options)
         if options.empty?
           shell_out!(*clean_array(*args))
         else
           shell_out!(*clean_array(*args), **options)
         end
       end
 
-      # helper sugar for resources that support passing timeouts to shell_out
+      def maybe_add_timeout(options)
+        if is_a?(Chef::Provider) && !new_resource.is_a?(Chef::Resource::LWRPBase) && new_resource.respond_to?(:timeout) && !options.key?(:timeout)
```

Comment:
Should probably check for `respond_to?(:new_resource)` before trying to call it.

Reply:
Chef::Provider objects all have new_resource's though.  i had that, but it seemed redundant...